modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/LazyExecutionContext.java

```diff
@@ -52,7 +53,7 @@ public LazyExecutionContext(ResolverSet resolverSet, M componentModel, Extension
   }
 
   private Map<String, LazyValue<Object>> getValueResolvers(ResolverSet resolverSet) {
-    Map<String, LazyValue<Object>> valueResolvers = new HashMap<>();
+    Map<String, LazyValue<Object>> valueResolvers = forSize(resolverSet.getResolvers().size());
```

Comment:
Would be nice to have a collector to a smallmap (see org.mule.runtime.api.util.collection.Collectors)